check.go

```diff
@@ -268,6 +287,29 @@ func (c *CheckRunner) UpdateCheck(checkID types.CheckID, status, output string)
 		return
 	}
 
+	// Do nothing if update is idempotent
+	if check.Status == status && check.Output == output {
+		check.failures = 0
+		check.passes = 0
+		return
+	}
+
+	if status == api.HealthCritical && c.MaxFailingChecks > 0 {
+		if check.failures > c.MaxFailingChecks {
+			check.failures = 0
+		} else {
+			check.failures++
+			return
+		}
+	} else if status != api.HealthCritical && c.MinPassingChecks > 0 {
+		if check.passes > c.MinPassingChecks {
+			check.passes = 0
+		} else {
+			check.passes++
+			return
+		}
+	}
```

Comment:
While this flow is completely correct, I can be easier to read if logic is inverter, i.e.
```go
if status == api.HealthPassing {
	successCounter++
	failuresCounter = 0
	if successCounter >= successBeforePassing {
		// Check successfull
		// ...		
		return
	}
	// Check passing but success threshold not reached
	// ...
} else {
	failuresCounter++
	successCounter = 0
	if failuresCounter >= failuresBeforeCritical {
			// Check failing
			// ...
			return
		}
		// Check failing but failure threshold not reached
		// ...
}

Reply:
I simplified the logic, it's similar to your proposal. Thanks!